### PR TITLE
[feature](merge-cloud) Add `SecurityChecker` for cloud

### PR DESCRIPTION
**Pull request by swjtu-zhanglei**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Comment by gavinchou** [comment]
LGTM

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Review by dataroaring — DISMISSED** [review]
LGTM

File: fe/fe-common/src/main/java/org/apache/doris/cloud/security/UrlSecurityChecker.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.lang.reflect.Method;
+
+public class UrlSecurityChecker extends SecurityChecker {
+    private static final Logger LOG = LogManager.getLogger(UrlSecurityChecker.class);
+
+    private static String checkerClassName = "com.aliyun.securitysdk.SecurityUtil";
```
Comment (gavinchou) [inline fe/fe-common/src/main/java/org/apache/doris/cloud/security/UrlSecurityChecker.java:30]: this is unnecessary.
Config.security_checker_class_name is enough.

File: fe/fe-common/src/main/java/org/apache/doris/cloud/security/UrlSecurityChecker.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+            }
+
+            Class clazz = Class.forName(checkerClassName);
+            jdbcUrlCheckMethod = clazz.getMethod("filterJdbcConnectionSource", String.class);
+            urlSecurityCheckMethod = clazz.getMethod("startSSRFNetHookChecking", String.class);
+            urlSecurityStopCheckMethod = clazz.getMethod("stopSSRFNetHookChecking", String.class);
+        } catch (Exception e) {
+            LOG.warn("exception:", e);
+            e.printStackTrace();
+            System.out.println("Failed to find com.aliyun.securitysdk.SecurityUtil's method");
```
Comment (gavinchou) [inline fe/fe-common/src/main/java/org/apache/doris/cloud/security/UrlSecurityChecker.java:50]: print Config.security_checker_class_name

**Review by gavinchou — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java [hunk 3]
```diff
@@ -428,13 +429,46 @@ private TNetworkAddress selectCloudRedirectBackend(String clusterName, String re
             throw new LoadException("Invalid header host: " + reqHost);
         }
 
-        if (InetAddressValidator.getInstance().isValid(reqHost)
-                && publicHostPort != null && reqHost == publicHostPort.first) {
-            return new TNetworkAddress(publicHostPort.first, publicHostPort.second);
-        } else if (privateHostPort != null) {
-            return new TNetworkAddress(reqHost, privateHostPort.second);
+        if (!Strings.isNullOrEmpty(Config.security_checker_class_name)) {
+            // ip
```
Comment (SWJTU-ZhangLei) [inline fe/fe-core/src/main/java/org/apache/doris/httpv2/rest/LoadAction.java:433]: 后面改个配置名区分不同逻辑

File: fe/fe-common/src/main/java/org/apache/doris/cloud/security/UrlSecurityChecker.java (hunk 1 above)
Comment (SWJTU-ZhangLei) [reply fe/fe-common/src/main/java/org/apache/doris/cloud/security/UrlSecurityChecker.java]: > this is unnecessary. Config.security_checker_class_name is enough.

resolved

**Comment by SWJTU-ZhangLei** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**